walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,59 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> times = new ArrayList<TimeRange>();
+    int beginning = TimeRange.START_OF_DAY;
+    int end = TimeRange.END_OF_DAY;
+    TimeRange middle_current;
+    TimeRange middle_past = TimeRange.fromStartDuration(TimeRange.END_OF_DAY, 0);
+    //check for primary edge cases of duration longer than one day
+    if (request.getDuration() > (24*60)){
```

Comment:
It will be better to cover these cases gracefully in the general algorithm instead of edge-casing them.